File: c++/dynamic_fail_2.cpp
// @expect error

#include <smack.h>//<cassert>

class Fish
{
public:
    virtual int numTails()
    {
        return 1;
    }
};

class TwinTailBeta: public Fish
{
public:
    int numTails()
    {
        return 2;
    }
};

int main(void)
{
    Fish *f = new Fish();
    Fish *b = new TwinTailBeta();
    
    assert(f->numTails() == 1);
    assert(b->numTails() != 2); 
    
    delete f;
    delete b;
}

File: swift/dynamic_fail_2.swift
// @expect error

class Fish {
  func numTails() -> Int {
    return 1
  }
}

class TwinTailBeta : Fish {
  override func numTails() -> Int {
    return 2
  }
}

let fish = Fish()
let beta = TwinTailBeta()

__VERIFIER_assert(fish.numTails() == 1)
__VERIFIER_assert(beta.numTails() != 2)
